src/v/cluster/topics_frontend.cc

```diff
@@ -385,6 +388,63 @@ make_allocation_request(const custom_assignable_topic_configuration& ca_cfg) {
     return req;
 }
 
+/**
+ * @param create_count how many new partitions are being created
+ * @param replica_count replication factor of the partitions
+ * @return true if we can accommodate the partitions
+ */
+bool topics_frontend::validate_partition_count(
```

Comment:
Can we simply change limit in partition allocator ? Allocation is done before replicating any data. Partition allocator has all necessary information to perform the validation while allocating partitions. It will simply return an error.

Reply:
We can add missing disk size, memory and fd limits to the allocator.

Reply:
+1